client/src/component/FlightSearchComponent/FlightSearch.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useRef } from 'react';
+
+import { Grid, Paper, TextField, Button, InputLabel } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+
+import useStyles from './styles';
+
+// Mock Data
+const cities = [
+  { title: 'Vancouver' },
+  { title: 'Calgary' },
+  { title: 'Toronto' },
+  { title: 'Bangkok' },
+];
+
+const FlightSearch = () => {
+  const classes = useStyles();
+  const ref0 = useRef();
+  const curr = new Date();
+  curr.setDate(curr.getDate());
+  const date = curr.toISOString().substr(0, 10);
+
+  const [state, setState] = useState({
+    from: 'Vancouver',
+    to: 'Bangkok',
+    arrival: date,
+    departure: date,
+    travellers: 1,
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setState((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const onSelect = (e, v, r) => {
```

Comment:
Same with the function parameters. More descriptive names go a long way to make the code more readable.
I also don't see you using the 'r' parameter anywhere in the function (or 'e' for that matter).